src/Discord.Net.Commands/CommandParser.cs

```diff
@@ -109,7 +109,12 @@ private enum ParserPart
                 if (argString != null)
                 {
                     if (curParam == null)
-                        return ParseResult.FromError(CommandError.BadArgCount, "The input text has too many parameters.");
+                    {
+                        if (commandService._failOnTooManyArgs)
```

Comment:
Personally, I'm more of a fan of passing a bool arg `failOnExtraArgs` instead of the whole CommandService.